robottelo/config/base.py

```diff
@@ -353,6 +354,71 @@ def validate(self):
         return validation_errors
 
 
+class ContainerRepositorySettings(FeatureSettings):
+    """Settings for syncing containers from container registries"""
+    section = 'container_repo'
+
+    repo_config_required = ['label', 'registry_url', 'registry_username',
+                            'registry_password', 'repos_to_sync']
+
+    def __init__(self, *args, **kwargs):
+        super(ContainerRepositorySettings, self).__init__(*args, **kwargs)
+        self.config_file = None
+        self.config = None
+        self.multi_registry_test_configs = None
+        self.yaml = None
+        self.long_pass_registry = None
+
+    def read(self, reader):
+        """Read container repo settings and associated yaml file"""
+        self.config_file = reader.get(self.section, 'config_file')
+        if self.config_file:
+            with open(self.config_file) as cf:
+                self.yaml = yaml.safe_load(cf)
+                self.config = self.yaml.get(self.section, None)
+                if self.config:
+                    self.long_pass_registry = self.config.get(
+                        'long_pass_test_registry', None)
+                    self.multi_registry_test_configs = self.config.get(
+                        'multi_registry_test_configs', None)
```

Comment:
With regards to the config file path, I image whenever you guys want to move over to using YAML for config, the option won't be required anymore, or it will be a different option.

I was debating whether to just read `robottelo.yaml` around the same time as `robottelo.properties`, but felt that it be better to do that during a whole sale move.

Reply:
I kept `self.config` since i needed it in `validate`

Reply:
@pgagne what do you think about his formatting change suggestions?

Reply:
@JacobCallahan I think this actually looks better (de-indented everything after the `with`statement)
```python
        if self.config_file:
            with open(self.config_file) as cf:
                self.yaml = yaml.safe_load(cf)
            self.config = self.yaml.get(self.section, None)
            if self.config:
                self.long_pass_registry = self.config.get(
                    'long_pass_test_registry', None)
                self.multi_registry_test_configs = self.config.get(
                    'multi_registry_test_configs', None)
```